frontend/packages/console-shared/src/selectors/common.ts

```diff
@@ -1,21 +1,21 @@
 
-export const hasLabel = (obj: K8sResourceKind, label: string): boolean =>
+export const hasLabel = (obj: K8sResourceCommon, label: string): boolean =>
   _.has(obj, ['metadata', 'labels', label]);
-export const getName = <A extends K8sResourceKind = K8sResourceKind>(value: A) =>
-  _.get(value, 'metadata.name') as K8sResourceKind['metadata']['name'];
-export const getNamespace = <A extends K8sResourceKind = K8sResourceKind>(value: A) =>
-  _.get(value, 'metadata.namespace') as K8sResourceKind['metadata']['namespace'];
-export const getUID = <A extends K8sResourceKind = K8sResourceKind>(value: A) =>
-  _.get(value, 'metadata.uid') as K8sResourceKind['metadata']['uid'];
-export const getDeletetionTimestamp = <A extends K8sResourceKind = K8sResourceKind>(value: A) =>
-  _.get(value, 'metadata.deletionTimestamp') as K8sResourceKind['metadata']['deletionTimestamp'];
-export const getCreationTimestamp = <A extends K8sResourceKind = K8sResourceKind>(value: A) =>
-  _.get(value, 'metadata.creationTimestamp') as K8sResourceKind['metadata']['creationTimestamp'];
-export const getAPIVersion = <A extends K8sResourceKind = K8sResourceKind>(value: A) =>
-  _.get(value, 'apiVersion') as K8sResourceKind['apiVersion'];
-export const getKind = <A extends K8sResourceKind = K8sResourceKind>(value: A) =>
-  _.get(value, 'kind') as K8sResourceKind['kind'];
-export const getOwnerReferences = <A extends K8sResourceKind = K8sResourceKind>(value: A) =>
-  _.get(value, 'metadata.ownerReferences') as K8sResourceKind['metadata']['ownerReferences'];
+export const getName = <A extends K8sResourceCommon = K8sResourceCommon>(value: A) =>
+  _.get(value, 'metadata.name') as K8sResourceCommon['metadata']['name'];
+export const getNamespace = <A extends K8sResourceCommon = K8sResourceCommon>(value: A) =>
+  _.get(value, 'metadata.namespace') as K8sResourceCommon['metadata']['namespace'];
+export const getUID = <A extends K8sResourceCommon = K8sResourceCommon>(value: A) =>
+  _.get(value, 'metadata.uid') as K8sResourceCommon['metadata']['uid'];
+export const getDeletetionTimestamp = <A extends K8sResourceCommon = K8sResourceCommon>(value: A) =>
+  _.get(value, 'metadata.deletionTimestamp') as K8sResourceCommon['metadata']['deletionTimestamp'];
+export const getCreationTimestamp = <A extends K8sResourceCommon = K8sResourceCommon>(value: A) =>
+  _.get(value, 'metadata.creationTimestamp') as K8sResourceCommon['metadata']['creationTimestamp'];
+export const getAPIVersion = <A extends K8sResourceCommon = K8sResourceCommon>(value: A) =>
+  _.get(value, 'apiVersion') as K8sResourceCommon['apiVersion'];
+export const getKind = <A extends K8sResourceCommon = K8sResourceCommon>(value: A) =>
+  _.get(value, 'kind') as K8sResourceCommon['kind'];
+export const getOwnerReferences = <A extends K8sResourceCommon = K8sResourceCommon>(value: A) =>
+  _.get(value, 'metadata.ownerReferences') as K8sResourceCommon['metadata']['ownerReferences'];
```

Comment:
When #3747 merges, we should be able to switch to optional chaining for better type safety.

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#optional-chaining